cmd/thanos/rule.go

```diff
@@ -472,49 +472,26 @@ func runRule(
 	}
 
 	// Handle reload and termination interrupts.
-	reload := make(chan struct{}, 1)
 	{
 		cancel := make(chan struct{})
-		reload <- struct{}{} // Initial reload.
-
 		g.Add(func() error {
+			// Initialize rules.
+			if err := reloadRules(logger, ruleFiles, ruleMgr, evalInterval, configSuccess, configSuccessTime, rulesLoaded); err != nil {
+				level.Error(logger).Log("msg", "initialize rules failed", "err", err)
+				// Return when initialize with invalid pattern error.
+				if _, ok := err.(*errInvalidPattern); ok {
```

Comment:
Consider using recently introduced errors API. https://blog.golang.org/go1.13-errors
```go
// Similar to:
//   if e, ok := err.(*QueryError); ok { … }
var e *QueryError
if errors.As(err, &e) {
    // err is a *QueryError, and e is set to the error's value
}
```

Reply:
Sure. Will look into it.

Reply:
@kakkoyun According to the coverstaion, I should not change the behavior the the rule reloading. This code will be removed.